File: src/基础算法和其他/模拟/PAT_Advanced_Level_Practice1117_EddingtonNumber/Main.cpp
#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;

int main(){
    int n;
    cin >> n;
    int arr[n];
    for (int i = 0; i < n; ++i) {
        cin >> arr[i];
    }
    sort(arr, arr + n);

    int ans = 1;
       for (int j = n - 1; j >= 0; --j, ++ans) {
           if (arr[j] <= ans) {
               std::cout << ans - 1 << std::endl;
               break;
           }
       }

    return 0;
}

File: src/基础算法和其他/模拟/PAT_Advanced_Level_Practice1117_EddingtonNumber/Main.java
package 基础算法和其他.模拟.PAT_Advanced_Level_Practice1117_EddingtonNumber;

import java.util.Arrays;
import java.util.Scanner;

public class Main {
    // E天的骑行距离超过E英里
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[] arr = new int[n];
        for(int i = 0; i < n; i++) arr[i] = scanner.nextInt();
        scanner.close();

        Arrays.sort(arr);

        for(int j = n - 1, ans = 1; j >= 0; j--, ans++){
            if(arr[j] <= ans){
                System.out.println(ans - 1);
                break;
            }
        }
    }
}
